ingress-controller/internal/manager/setup.go

```diff
@@ -374,10 +368,10 @@ func AdminAPIClientFromServiceDiscovery(
 		retry.Delay(delay),
 		retry.OnRetry(func(_ uint, err error) {
 			// log the error if the error is NOT caused by 0 available gateway endpoints.
-			if !errors.As(err, &NoAvailableEndpointsError{}) {
-				logger.Error(err, "Failed to create kong client(s)")
+			if !errors.As(err, &ingresserrors.NoAvailableEndpointsError{}) {
+				logger.V(logging.DebugLevel).Info("Failed to create kong client(s)", "error", err)
```

Comment:
I think this way this will be logged twice, how about adding a return here to prevent that?

```suggestion
				logger.V(logging.DebugLevel).Info("Failed to create kong client(s)", "error", err)
				return
```

Reply:
I tried to not modify the existing code, but yeah, I can do that.